components/org.wso2.carbon.identity.api.server.action.management/org.wso2.carbon.identity.api.server.action.management.v1/src/main/java/org/wso2/carbon/identity/api/server/action/management/v1/impl/ActionsApiServiceImpl.java

```diff
@@ -39,8 +39,15 @@
  */
 public class ActionsApiServiceImpl implements ActionsApiService {
 
-    @Autowired
-    ServerActionManagementService serverActionManagementService;
+    private final ServerActionManagementService serverActionManagementService;
+
+    public ActionsApiServiceImpl() {
+        try {
+            this.serverActionManagementService = ActionManagementServiceFactory.getActionManagementService();
+        } catch (Exception e) {
```

Comment:
Have thrown the generic Exception type from the ActionManagementServiceFactory.getActionManagementService(); method?